devel/include/detail/wtr/watcher/adapter/windows/watch.hpp

```diff
@@ -175,7 +180,22 @@ inline auto do_event_send(
           }
         }();
 
-        callback({path_name, effect_type, path_type});
+        if (buf->Action == FILE_ACTION_RENAMED_OLD_NAME)
+        {
+          old_path_name = path_name;
+          old_effect_type = effect_type;
+          old_path_type = path_type;
+        }
+        else if (buf->Action == FILE_ACTION_RENAMED_NEW_NAME)
+        {
+          ::wtr::watcher::event old_event{old_path_name, old_effect_type, old_path_type};
+          ::wtr::watcher::event new_event{path_name, effect_type, path_type};
+          callback({old_event, std::move(new_event)});
+        }
```

Comment:
Are these guaranteed to come one after the other, in this order, or is it just likely that it happens this way?

Wondering if we need a bit more logic to reorder these events if they don't come in this expected order

Reply:
Good question.  I saw some comments suggesting as much and I also tested and saw that behavior but it's possible it wouldn't?  At least I don't recall seeing anything in the documentation, I'll look again later.

Reply:
Sorry for the delay @e-dant .  It's a bit verbose but I tested this successfully and should now work regardless of order.  Is there a reason the builders fail?

Reply:
Builders fail because of some issues around GitHub tokens, which I haven't debugged yet.

That said, our windows unit tests, when run on GitHub's CI, have always been weird. I've had different results on actual hardware.

Reply:
The most recent commit looks great from a functional perspective